.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,212 @@
+name: Build and Test
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    name: build:required
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup PNPM
+        uses: dfinity/ci-tools/actions/setup-pnpm@main
+        with:
+          node_version_file: '.nvmrc'
+
+      - name: Build
+        run: pnpm build
+
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-artifact
+          path: ./packages/*/lib/
+
+  tests:
+    name: tests:required
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        test_type: [unit, e2e, mitm]
+        node:
+          - 20.18 # without require(esm)
+          - 20.19 # with require(esm), see https://nodejs.org/en/blog/release/v20.19.0/#requireesm-is-now-enabled-by-default
+          - 22
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup PNPM
+        uses: dfinity/ci-tools/actions/setup-pnpm@main
+        with:
+          node_version: ${{ matrix.node }}
+
+      - name: Download Build Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-artifact
+          path: ./
+
+      - name: Setup DFX
+        if: ${{ matrix.test_type == 'mitm' || matrix.test_type == 'e2e' }}
+        uses: dfinity/setup-dfx@main
+
+      - name: Setup e2e tests
```

Comment:
I wanna get this to a point where `pnpm test` just runs everything and there's none of these project-specific scripts.